internal/preferences.go

```diff
@@ -270,9 +271,26 @@ func (p *InMemoryPreferences) remove(key string) {
 func (p *InMemoryPreferences) set(key string, value interface{}) {
 	p.lock.Lock()
 
-	if stored, ok := p.values[key]; ok && stored == value {
-		p.lock.Unlock()
-		return
+	if reflect.TypeOf(value).Kind() == reflect.Slice {
+		s := reflect.ValueOf(value)
+		old := reflect.ValueOf(p.values[key])
+		if p.values[key] != nil && s.Len() == old.Len() {
+			changed := false
+			for i := 0; i < s.Len(); i++ {
+				if s.Index(i).Interface() != old.Index(i).Interface() {
+					changed = true
```

Comment:
Would it not be wise to break out of the loop once we find that it has changed?
```suggestion
					changed = true
					break
```